pennylane_cirq/simulator_device.py

```diff
@@ -136,30 +135,36 @@ def apply(self, operations, **kwargs):
         self.circuit.append(cirq.IdentityGate(len(self.qubits))(*self.qubits))
 
         if self.analytic:
-            self._result = self._simulator.simulate(
-                self.circuit, initial_state=self._initial_state
-            )
+            self._result = self._simulator.simulate(self.circuit, initial_state=self._initial_state)
 
-            self._state = np.array(self._result.state_vector())
+            self._state = self._get_state_from_cirq(self._result)
 
     def analytic_probability(self, wires=None):
         # pylint: disable=missing-function-docstring
         if self._state is None:
             return None
 
         wires = wires or range(self.num_wires)
-        probs = np.abs(self._state) ** 2
+        probs = self._get_computational_basis_probs()
 
         return self.marginal_prob(probs, wires)
 
+    def _get_state_from_cirq(self, result):
+        """Extract the state array from a Cirq TrialResult ``result``"""
+        return np.array(result.state_vector())
+
+    def _get_computational_basis_probs(self):
+        """Extract the probabilities of all computational basis measurements."""
+        return np.abs(self._state) ** 2
+
     @property
     def state(self):
         """Returns the state vector of the circuit prior to measurement.
 
         .. note::
 
             The state includes possible basis rotations for non-diagonal 
-            observables. Note that this behaviour is differs from PennyLane's
+            observables. Note that this behaviour differs from PennyLane's
```

Comment:
Note: I wasn't able to understand why there should be a difference between this plugin's behaviour and `default.qubit`'s. I suspect it was just left as a "todo" for future improvement

Reply:
See my comment below, it can be done by invoking `simulate` twice.